resolwe/flow/execution_engines/workflow/__init__.py

```diff
@@ -0,0 +1,106 @@
+        if steps is None:
+            return
+
+        if not isinstance(steps, list):
+            raise ExecutionError('Workflow program must be a list of steps.')
+
+        # Expression engine evaluation context.
+        context = {
+            'inputs': {},
+            'steps': collections.OrderedDict(),
+        }
+
+        for index, step in enumerate(steps):
+            try:
+                step_id = step['id']
+                step_slug = step['run']
+            except KeyError as error:
+                raise ExecutionError('Incorrect definition of step "{}", missing property "{}".'.format(
+                    step.get('id', index), error
+                ))
+
+            # Fetch target process.
+            process = Process.objects.filter(slug=step_slug).order_by('-version').first()
+            if not process:
+                raise ExecutionError('Incorrect definition of step "{}", invalid process "{}".'.format(
+                    step_id, step_slug
+                ))
+
+            # Process all input variables.
+            step_input = step.get('input', {})
+            if not isinstance(step_input, dict):
+                raise ExecutionError('Incorrect definition of step "{}", input must be a dictionary.'.format(
+                    step_id
+                ))
+
+            data_input = {}
+            for input_name, input_value in step_input.items():
+                if isinstance(input_value, string_types):
+                    input_value = input_value.strip()
+                    if expression_engine is not None:
+                        # TODO: Should this be expression-engine dependent?
+                        try:
+                            if input_value.startswith('{{') and input_value.endswith('}}'):
+                                # Inline expression.
+                                input_value = expression_engine.evaluate_inline(input_value[2:-2], context)
+                            else:
+                                # Block expression.
+                                input_value = expression_engine.evaluate_block(input_value, context)
+                        except EvaluationError as error:
+                            raise ExecutionError('Error while evaluating expression for step "{}":\n{}'.format(
+                                step_id, error
+                            ))
+
+                data_input[input_name] = input_value
+
+            # Create the data object.
+            data_object = Data.objects.create(
+                process=process,
+                contributor=data.contributor,
+                input=data_input,
+            )
+
+            # TODO: What about permissions?
+
+            context['steps'][step_id] = data_object.pk
+
+        # Immediately set our status to done and output all data object identifiers.
+        data.output = {
+            'steps': list(context['steps'].values()),
+        }
+        data.status = Data.STATUS_DONE
```

Comment:
Yes, but in any case, we need support for tracking child processes. Perhaps in a similar way to how we already track dependencies between data objects (through inputs), but here the execution engines would explicitly declare some data objects as being children. What do you think?